blockchain/state_manager/src/lib.rs

```diff
@@ -41,23 +42,45 @@ where
             .ok_or_else(|| Error::ActorStateNotFound(actor.state.to_string()))?;
         Ok(act)
     }
-    /// Returns the epoch at which the miner was slashed at
-    pub fn miner_slashed(&self, _addr: &Address, _ts: &Tipset) -> Result<u64, Error> {
-        // TODO update to use power actor if needed
-        todo!()
+    fn get_actor(&self, addr: &Address, state_cid: &Cid) -> Result<Option<ActorState>, Error> {
+        let state =
+            HamtStateTree::new_from_root(self.bs.as_ref(), state_cid).map_err(Error::State)?;
+        state.get_actor(addr).map_err(Error::State)
     }
-    /// Returns the amount of space in each sector committed to the network by this miner
-    pub fn miner_sector_size(&self, addr: &Address, ts: &Tipset) -> Result<SectorSize, Error> {
-        let act: miner::State = self.load_actor_state(addr, ts)?;
-        // * Switch back to retrieving from Cid if/when changed in actors
-        // let info: miner::MinerInfo = self.bs.get(&act.info)?.ok_or_else(|| {
-        //     Error::State("Could not retrieve miner info from IPLD store".to_owned())
-        // })?;
-        Ok(act.info.sector_size)
+    /// Returns true if miner has been slashed or is considered invalid
+    pub fn is_miner_slashed(&self, addr: &Address, state_cid: &Cid) -> Result<bool, Error> {
+        let ms: miner::State = self.load_actor_state(addr, state_cid)?;
+        if ms.post_state.has_failed_post() {
+            return Ok(true);
+        }
+
+        let ps: power::State = self.load_actor_state(&*STORAGE_POWER_ACTOR_ADDR, state_cid)?;
+        match ps.get_claim(self.bs.as_ref(), addr)? {
+            Some(_) => Ok(false),
+            None => Ok(true),
+        }
     }
-    pub fn get_actor(&self, addr: &Address, ts: &Tipset) -> Result<Option<ActorState>, Error> {
-        let state = HamtStateTree::new_from_root(self.bs.as_ref(), ts.parent_state())
-            .map_err(Error::State)?;
-        state.get_actor(addr).map_err(Error::State)
+    /// Returns raw address of a miner
+    pub fn get_miner_wrk_addr(&self, state_cid: &Cid, addr: &Address) -> Result<Address, Error> {
```

Comment:
can we not use short form for at least `work`? I don't see the need for this to be that shortened

Reply:
lol just to clarify, you do not want the use of short-form verbiage? I have no objections to that and I am happy to change. However, we do have many function signatures that have similar short-form usages so we should probably enforce it across the board?

Reply:
I think what @austinabell means here specfically is that were only shortening it by 1 char. Doesn't make much sense to shorten it. Might as well shorten miner to mnr 😂  go full startup company name (dropping all the vowels)

Reply:
We're trendy haha

Reply:
yeah guess I meant to prefix the comment with optional, was just confusing for me to read